backend/remote-state/http/backend.go

```diff
@@ -0,0 +1,346 @@
+	if b.client != nil {
+		return nil
+	}
+	// Grab the resource data
+	data := schema.FromContextBackendConfig(ctx)
+
+	// Store the URL address
+	addressURL := data.Get("address").(string)
+	// Validate URL address
+	validationErr := assertValidURL(addressURL)
+	if validationErr != nil {
+		return validationErr
+	}
+	b.address = addressURL
+
+	b.updateMethod = data.Get("update_method").(string)
+
+	if v, ok := data.GetOk("lock_address"); ok {
+		lockAddress := v.(string)
+		// Validate lockAddress
+		validationErr := assertValidURL(lockAddress)
+		if validationErr != nil {
+			return validationErr
+		}
+		b.lockAddress = lockAddress
+	} else {
+		// If lockAddress is null, use the http rest api address
+		b.lockAddress = b.address
+	}
+
+	b.lockMethod = data.Get("lock_method").(string)
+
+	if v, ok := data.GetOk("unlock_address"); ok {
+		unlockAddress := v.(string)
+		// Validate unlockAddress
+		validationErr := assertValidURL(unlockAddress)
+		if validationErr != nil {
+			return validationErr
+		}
+		b.unlockAddress = unlockAddress
+	} else {
+		// If unlockAddress is null, use the http rest api address
+		b.unlockAddress = b.address
+	}
+
+	b.unlockMethod = data.Get("unlock_method").(string)
+
+	if v, ok := data.GetOk("username"); ok {
+		b.username = v.(string)
+	}
+
+	if v, ok := data.GetOk("password"); ok {
+		b.password = v.(string)
+	}
+
+	client := &http.Client{
+		Timeout: time.Second * 10,
+	}
+
+	if v, ok := data.GetOk("skip_cert_verification"); ok {
+		b.skipTLS = v.(bool)
+		if b.skipTLS {
+			// If skip_cert_verification = true, the address must be of type HTTPS
+			if !isHTTPS(addressURL) {
+				return fmt.Errorf("Address must be of type HTTPS if skip_cert_verification = true")
+			}
+			// If local_cert_ca_file is also set, raise an error
+			if v, ok := data.GetOk("local_cert_ca_file"); ok {
+				return fmt.Errorf("skip_cert_verification is %t and local_cert_ca_file is set: %s. please choose one or the other", b.skipTLS, v)
+			}
+			// If mutual_tls_authentication is also set, raise an error
+			if data.Get("mutual_tls_authentication").(bool) == true {
+				return fmt.Errorf("skip_cert_verification is true and mutual_tls_authentication is set. please choose one or the other")
+			}
+			// Replace the client with one that ignores TLS verification
+			client = &http.Client{
+				Timeout: time.Second * 10,
+				Transport: &http.Transport{
+					TLSClientConfig: &tls.Config{
+						InsecureSkipVerify: true,
+					},
+				},
+			}
+		}
+	}
+
+	if v, ok := data.GetOk("local_cert_ca_file"); ok {
+		// If local_cert_ca_file exists, the address must be of type HTTPS
+		if !isHTTPS(addressURL) {
+			return fmt.Errorf("Address must be of type HTTPS if local_cert_ca_file is set")
+		}
+		if data.Get("mutual_tls_authentication").(bool) == true {
+			return fmt.Errorf("mutual_tls_authentication is true and local_cert_ca_file is set. Please choose one or the other")
+		}
+		if data.Get("skip_cert_verification").(bool) == true {
```

Comment:
I think you can omit this check. Haven't you already asserted that `skip_cert_verification` is false in the block on L200? If `skip_cert_verification` is true, and `local_ca_cert_file` is set, you already error out on L208.